File: pdu/python/hako_srv_msgs/pdu_cdr_conv_SystemControlResponsePacket.py
from .pdu_pytype_SystemControlResponsePacket import SystemControlResponsePacket
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..hako_srv_msgs.pdu_cdr_conv_ServiceResponseHeader import *
from ..hako_srv_msgs.pdu_cdr_conv_SystemControlResponse import *



def py_to_cdr_body_SystemControlResponsePacket(writer: CdrWriter, src: SystemControlResponsePacket):
    py_to_cdr_body_ServiceResponseHeader(writer, src.header)
    py_to_cdr_body_SystemControlResponse(writer, src.body)


def cdr_body_to_py_SystemControlResponsePacket(reader: CdrReader, dst: SystemControlResponsePacket):
    cdr_body_to_py_ServiceResponseHeader(reader, dst.header)
    cdr_body_to_py_SystemControlResponse(reader, dst.body)
    return dst


def py_to_cdr_SystemControlResponsePacket(src: SystemControlResponsePacket) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_SystemControlResponsePacket(writer, src)
    return writer.bytes()


def cdr_to_py_SystemControlResponsePacket(cdr_payload) -> SystemControlResponsePacket:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = SystemControlResponsePacket()
    return cdr_body_to_py_SystemControlResponsePacket(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_SystemControlResponsePacket.js
import { SystemControlResponsePacket } from './pdu_jstype_SystemControlResponsePacket.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { ServiceResponseHeader } from '../hako_srv_msgs/pdu_jstype_ServiceResponseHeader.js';
import { PduServiceResponseHeaderConverter } from '../hako_srv_msgs/pdu_cdr_conv_ServiceResponseHeader.js';
import { SystemControlResponse } from '../hako_srv_msgs/pdu_jstype_SystemControlResponse.js';
import { PduSystemControlResponseConverter } from '../hako_srv_msgs/pdu_cdr_conv_SystemControlResponse.js';


export class PduSystemControlResponsePacketConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { SystemControlResponsePacket } src
     */
    static to_cdr_body(writer, src) {
        PduServiceResponseHeaderConverter.to_cdr_body(writer, src.header);
        PduSystemControlResponseConverter.to_cdr_body(writer, src.body);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { SystemControlResponsePacket } dst
     * @returns { SystemControlResponsePacket }
     */
    static cdr_body_to_js(reader, dst) {
        PduServiceResponseHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduSystemControlResponseConverter.cdr_body_to_js(reader, dst.body);
        return dst;
    }

    /**
     * @param { SystemControlResponsePacket } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { SystemControlResponsePacket }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new SystemControlResponsePacket());
    }
}
